common/common.cpp

```diff
@@ -1774,7 +1774,9 @@ void dump_kv_cache_view_seqs(const llama_kv_cache_view & view, int row_size) {
             if (cs_curr[j] < 0) { continue; }
             if (seqs.find(cs_curr[j]) == seqs.end()) {
                 if (seqs.size() + 1 >= sizeof(slot_chars)) { break; }
-                seqs[cs_curr[j]] = seqs.size();
+                seqs[cs_curr[j]] = 0;
+                const auto size = seqs.size();
```

Comment:
are you sure you are not changing behavoir here? now it first resizes and then inserts the new size. Before it was either undefined or first took the size and then resized.

Reply:
modified godbolt from the so you linked: https://godbolt.org/z/rbndMGqva
Here you can see that with gcc 7.1 you would change the behavior. I suspect almost no-one is using something as old as gcc6.x (gcc6.5 was released 2018)

Reply:
@Green-Sky,
you mean fix it on this code with integer literal?
```cpp
seqs[cs_curr[j]] = 0_z;
```

Reply:
I meant the order of `seqs.size()` and `seqs[]`. To me it looks like your patch makes `.size()` now return a value larger by one.